test/utils/OrderPayable.sol.js

```diff
@@ -0,0 +1,319 @@
+            const orderSignerAddress = roles.manager.address;
+
+            const network = await orderPayable.provider.getNetwork();
+            const chainId = network.chainId;
+
+            const hashedMessage = ethers.utils.solidityKeccak256(
+              ['uint256', 'address', 'bytes32', 'uint256', 'uint256'],
+              [chainId, orderPayable.address, orderId, expirationTimestamp, paymentAmount]
+            );
+
+            const hash = ethers.utils.arrayify(hashedMessage);
+
+            const signature = await roles.manager.signMessage(ethers.utils.arrayify(hash));
+
+            const encodedData = ethers.utils.defaultAbiCoder.encode(
+              ['bytes32', 'uint256', 'address', 'bytes'],
+              [orderId, expirationTimestamp, orderSignerAddress, signature]
+            );
+
+            await expect(
+              orderPayable.connect(roles.manager).payForOrder(encodedData, { value: paymentAmount })
+            ).to.be.revertedWith('Order expired');
+            expect(await ethers.provider.getBalance(orderPayable.address)).to.equal(ethers.utils.parseEther('0'));
+            expect(await orderPayable.orderIdToPaymentStatus(orderId)).to.equal(false);
+          });
+        });
+      });
+      context('Order Id zero', function () {
+        it('Target function reverts', async function () {
+          const { roles, orderPayable } = await deploy();
+
+          const orderId = ethers.utils.hexZeroPad(
+            '0x0000000000000000000000000000000000000000000000000000000000000000',
+            32
+          );
+          const timestamp = await helpers.time.latest();
+          const expirationTimestamp = timestamp + 60;
+          const paymentAmount = ethers.utils.parseEther('1');
+          const orderSignerAddress = roles.manager.address;
+
+          const network = await orderPayable.provider.getNetwork();
+          const chainId = network.chainId;
+
+          const hashedMessage = ethers.utils.solidityKeccak256(
+            ['uint256', 'address', 'bytes32', 'uint256', 'uint256'],
+            [chainId, orderPayable.address, orderId, expirationTimestamp, paymentAmount]
+          );
+
+          const hash = ethers.utils.arrayify(hashedMessage);
+
+          const signature = await roles.manager.signMessage(ethers.utils.arrayify(hash));
+
+          const encodedData = ethers.utils.defaultAbiCoder.encode(
+            ['bytes32', 'uint256', 'address', 'bytes'],
+            [orderId, expirationTimestamp, orderSignerAddress, signature]
+          );
+
+          await expect(
+            orderPayable.connect(roles.manager).payForOrder(encodedData, { value: paymentAmount })
+          ).to.be.revertedWith('Order ID zero');
+          expect(await ethers.provider.getBalance(orderPayable.address)).to.equal(ethers.utils.parseEther('0'));
+          expect(await orderPayable.orderIdToPaymentStatus(orderId)).to.equal(false);
+        });
+      });
+    });
+  });
+
+  describe('withdraw', function () {
```

Comment:
You should test the case where the value transfer reverts. You can do that by setting the `recipient` to be a contract that doesn't have a `receive()` function (for example AccessControlRegistry).